manim/utils/file_ops.py

```diff
@@ -121,6 +122,16 @@ def write_to_movie() -> bool:
     )
 
 
+def ensure_executable(path_to_exe: Path) -> bool:
+    if path_to_exe.parent == Path("."):
+        executable = shutil.which(path_to_exe.stem)
+        if executable is None:
+            return False
+    else:
+        executable = path_to_exe
+    return os.access(executable, os.X_OK)
```

Comment:
How about running a simple version test to make sure that FFmpeg is actually working? It can be done in another PR though.

Reply:
> Checking ffmpeg -version every time manim runs seems excessive and bad for performance (but I don't have a strong opinion on it, maybe I overestimate the effect).

It would be better to check just before using it and not when importing manim. Like in the `scene_file_writer`.

Reply:
> How about running a simple version test to make sure that FFmpeg is actually working? It can be done in another PR though.

I mean changing the `path` key in the config file should be enough, to test whether the system works as expected.

Which leads to the next question, what should the `path` key be renamed to?

> Maybe if we can cache the result of the version check somehow? Checking ffmpeg -version every time manim runs seems excessive and bad for performance

I think we have more serious performance problems than that 😅. But, ideally we would want this check to be integrated into the installation step. Like a post install thing in using setuptools. But that requires a bigger change.

Reply:
That sounds reasonable. Does poetry have a way to define a post-install check? Could you open a new issue for that?

Reply:
There are some ways that I have come across on the web, but haven't delved too deep into it.
Sure, I can make an issue about it.

Reply:
> How about running a simple version test to make sure that FFmpeg is actually working? It can be done in another PR though.

This think this is the same thing as running a version check.